### ISSUE TITLE
Reduce amount of logs on element crash

### ISSUE THREAD

**Issue by mat-hek** [membraneframework/membrane_core#318]
We should remove these which are no longer useful, especially regarding warnings.

**Comment by sax** [comment]
Some thoughts about this:

There are a log of warnings when an element in a pipeline or bin crashes, and there are other elements and bins. This is from here: https://github.com/membraneframework/membrane_core/blob/af554c2098d43083e671e2cfceb17add85932c94/lib/membrane/core/element/lifecycle_controller.ex#L69-L77

I was thinking that one solution to this is that processes can be grouped into the same crash group. When this is the case, the shutdown reason is `{:shutdown, :membrane_crash_group_kill}`. If an element is shut down for this reason, `membrane_core` could debug rather than warn.

```elixir
if playback_state == :terminating || reason == {:shutdown, :membrane_crash_group_kill} do
  Membrane.Logger.debug("Terminating element, reason: #{inspect(reason)}")
else
  Membrane.Logger.warn("""
  Terminating element possibly not prepared for termination as it was in state #{inspect(playback_state)}.
  Reason: #{inspect(reason)}",
  State: #{inspect(state, pretty: true)}
  """)
end
```

This puts a big pressure on plugins to ensure that children are started in the same crash group. This is a little difficult right now, because it does not seem to me that crash group is available in `handle_init` or in other callbacks.

A quick, messy solution would be to change plugins like `membrane_rtp_plugin` and `membrane_webrtc_plugin` to add a `def_option` for `:crash_group_id`, which can be stored in state and used whenever any of the bins start any children.

A bigger change, that might be more controversial, would be for membrane core to start children in the same crash group as their parent unless specifically given a different crash group. That would be less overhead for plugin maintainers—I don't know if this would have unintended consequences, however.

**Comment by sax** [comment]
Probable consequences of having membrane core automatically propagate crash groups to children:

* Bins will need the `handle_crash_group_down/3` optional callback added to them
* When stopping processes in a crash group, they will need to be stopped in a bottom-up fashion, so that nested bins will not be stopped before their elements are stopped. Otherwise nested elements may receive a `{:shutdown, :parent_crash}` before they receive a `{:shutdown, :membrane_crash_group_kill}`.

For the latter issue, maybe when stopping processes in a crash group, Elements should be stopped before Bins?

https://github.com/membraneframework/membrane_core/blob/af554c2098d43083e671e2cfceb17add85932c94/lib/membrane/core/parent/child_life_controller.ex#L210-L223

**Comment by sax** [comment]
Another possibility is that when a Bin is shutdown, it can catch the fact that it's a crash group and try to gracefully shut down its children?

https://github.com/membraneframework/membrane_core/blob/af554c2098d43083e671e2cfceb17add85932c94/lib/membrane/core/bin.ex#L173-L176

In this case, maybe we would not need to automatically propagate the crash group through the system and keep track of it. Maybe this could be handled entirely via terminate callbacks?

**Comment by sax** [comment]
Here is a sketch of what this could look like: https://github.com/membraneframework/membrane_core/compare/master...geometerio:auto-crash-groups

There's a failing test having to do specifically with bins and crash groups, but we could fix that if this looks promising.

**Comment by sax** [comment]
Here's a much simpler approach: #322

**Comment by mat-hek** [comment]
closed by #322
